tests/test_run.py

```diff
@@ -9,6 +11,85 @@
 from tests.fakes import RecordingExecutor
 
 
+@pytest.mark.parametrize(
+    "synchronous_transitions",
+    [
+        [],
+        [ExecutorState.PREPARING, ExecutorState.FINALIZING],
+    ],
+)
+def test_handle_job_full_execution(synchronous_transitions, db, freezer):
+    # move to a second boundary so rounding doesn't affect our times.
```

Comment:
wondering if there's a way of wording this that doesn't make me initially wonder where the first boundary is 😆

Reply:
I'm sure there is :)

Reply:
```suggestion
    # move to a whole second boundary for easier timestamp maths
```